lib/lighthouse/service_exception.rb

```diff
@@ -27,9 +27,11 @@ class ServiceException
     def self.send_error(error, service_name, lighthouse_client_id, url)
       raise error_class(:'504') if gateway_timeout?(error.response)
 
-      error_response = error.response
+      error_response = error.response.deep_symbolize_keys
 
-      return error unless error_response&.key?(:status)
+      unless error_response&.key?(:status) || (error_response.instance_of?(Hash) && error_response&.key?(:status))
```

Comment:
This code feels overly defensive. `deep_symbolize_keys` is a method on hash that returns another hash, so I don't see why we need to confirm that it's a hash or use the safety operator because I don't see how it can return nil. Also, the second condition here can never be matched because it contains the first condition, so either the first condition is true and the second is never evaluated or they're both false. Maybe I'm missing something about how the response object works.

Reply:
Yeah good catch thanks. This probably came from some mixing/matching during the refactor. I'll fix this up.